### PR TITLE
adds migrate command and ansible base image scaffolds

### PR DESCRIPTION
**Pull request by mhrivnak**
The `operator-sdk migrate` command adds go source and other files necessary to
convert an ansible operator to a hybrid operator.

The scaffolds used for `migrate` are also made available for use in creating
the ansible-operator base image through
`commands/ansible-operator-base/main.go`.

### CODE REVIEW

**Comment by mhrivnak** [comment]
I'll add docs for this, but want to vet the general approach first, and also ensure we are happy enough with the contents of `main.go` that we're ready to start recommending that people use this.

File: hack/tests/e2e-ansible.sh [hunk 1]
```diff
@@ -1,99 +1,134 @@
...
+
+    # The deployment should get garbage collected, so we expect to fail getting the deployment.
+    if ! timeout 20s bash -c -- "while kubectl get deployment ${memcached_deployment} 2> /dev/null; do sleep 1; done";
+    then
+        echo FAIL: memcached Deployment did not get garbage collected
+        kubectl logs deployment/memcached-operator
+        exit 1
+    fi
+
+    ## TODO enable when this is fixed: https://github.com/operator-framework/operator-sdk/issues/818
```
Comment (dymurray) [inline hack/tests/e2e-ansible.sh]: I believe this can be added back. I've commented on the issue.

File: hack/tests/e2e-ansible.sh [hunk 2]
```diff
@@ -1,99 +1,134 @@
...
-    kubectl logs deployment/memcached-operator
-    exit 1
-fi
-memcached_deployment=$(kubectl get deployment -l app=memcached -o jsonpath="{..metadata.name}")
-if ! timeout 1m kubectl rollout status deployment/${memcached_deployment};
-then
-    kubectl logs deployment/${memcached_deployment}
-    exit 1
-fi
+deploy_operator
```
Comment (dymurray) [inline hack/tests/e2e-ansible.sh]: +1 to wrapping all this in functions.

File: hack/tests/e2e-ansible.sh [hunk 3]
```diff
@@ -1,99 +1,134 @@
...
-    exit 1
-fi
+operator-sdk migrate
 
-# The deployment should get garbage collected, so we expect to fail getting the deployment.
-if ! timeout 20s bash -c -- "while kubectl get deployment ${memcached_deployment}; do sleep 1; done";
+if [[ ! -e build/Dockerfile.sdkold ]];
 then
-    kubectl logs deployment/memcached-operator
+    echo FAIL the old Dockerful should have been renamed to Dockerfile.sdkold
```
Comment (AlexNPavel) [inline hack/tests/e2e-ansible.sh]:
```suggestion
    echo FAIL the old Dockerfile should have been renamed to Dockerfile.sdkold
```

**Comment by mhrivnak** [comment]
I am going to submit a new PR that takes a different approach to building the ansible operator binary itself. As discussed elsewhere, it will add a `run` command to the operator-sdk binary.

The `migrate` behavior will be the same as in this PR.

**Comment by mhrivnak** [comment]
Here's the new PR that obsoletes this one: https://github.com/operator-framework/operator-sdk/pull/887

**Pull request closed without merging**